src/controllers/update/company.profile.controller.ts

```diff
@@ -0,0 +1,52 @@
+import { Session } from "@companieshouse/node-session-handler";
+import { NextFunction, Request, Response } from "express";
+import { ApplicationData } from "model";
+import { logger } from "../../utils/logger";
+import * as config from "../../config";
+import { getApplicationData, setExtraData } from "../../utils/application.data";
+import { OeErrorKey } from "../../model/data.types.model";
+import { mapCompanyProfileToOverseasEntityToDTOUpdate } from "../../utils/update/company.profile.mapper.to.oversea.entity";
+import { CompanyProfile } from "@companieshouse/api-sdk-node/dist/services/company-profile/types";
+import { getCompanyProfile } from "../../service/company.profile";
+
+export const get = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    logger.debugRequest(req, `GET ${config.CONFIRM_OVERSEAS_ENTITY_DETAILS_PAGE}`);
+    const session = req.session as Session;
+    const appData: ApplicationData = getApplicationData(session);
+    const id: string | any = appData?.oe_number;
+    const companyDataResponse = await getCompanyProfile(req, id) as CompanyProfile;
+    const overseasEntity = mapCompanyProfileToOverseasEntityToDTOUpdate(companyDataResponse);
```

Comment:
You should only call mapCompanyProfileToOverseasEntity() after you check that companytDataResonse is missing. Then you don't need to protect against something invalid being passed in. Move line 19 to after line 22.